### ISSUE TITLE
Crash in brave_ads::DoesSupportUrl(GURL const&)

### ISSUE THREAD

**Issue by aseren** [brave/brave-browser#31033]
https://brave.sp.backtrace.io/p/brave/debug?filters=W1siY2FsbHN0YWNrIiwiY29udGFpbnMiLCJicmF2ZV9hZHM6OkRvZXNTdXBwb3J0VXJsIl0sWyJjYWxsc3RhY2siLCJjb250YWlucyIsImJyYXZlX2FkcyJdXQ%3D%3D&debug=(%224ecced2%22,0,2)

**Comment by tmancey** [comment]
Set as a P3 because the catalog needs an invalid URL for the crash to occur. cc @tackley

**Comment by aseren** [comment]
The crash is occurring in version 1.56.8.

https://brave.sp.backtrace.io/p/brave/debug?time=quarter&filters=W1siZmluZ2VycHJpbnQ7aXNzdWVzO3N0YXRlIiwiZXF1YWwiLCJvcGVuIl0sWyJjYWxsc3RhY2siLCJjb250YWlucyIsImJyYXZlX2FkcyJdXQ%3D%3D&fingerprint=46e6eefb716fba421cdcdff0345077f93218c259611b038378e6ef30d73edda5&debug=(%2253d9d82%22,0,0)

**Comment by tmancey** [comment]
Crashes would only occur for very old browser versions and there are few crashes that would bring down the utility process, so in my opinion we do not need to uplift because the fix is in 1.57.x in preparation for search result.

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.57.27 Chromium: 115.0.5790.114 (Official Build) beta (64-bit) 
-- | --
Revision | 18977fb43f693d5a8deeb48bd1cfa52651d3f73e
OS | Ubuntu 18.04 LTS

Verified test plan from https://github.com/brave/brave-core/pull/18895


Verified ad with invalid url is ignored:
```
[26631:26631:0731/112352.927740:VERBOSE1:eligible_notification_ads_v2.cc(38)] Get eligible notification ads
[26631:26631:0731/112352.932566:VERBOSE1:eligible_notification_ads_v2.cc(102)] No eligible ads

```

Verified no crash
